components/kyma-environment-broker/cmd/deprovisionretrigger/main.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/events"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage/dbmodel"
+	"github.com/kyma-project/control-plane/components/schema-migrator/cleaner"
+	log "github.com/sirupsen/logrus"
+	"github.com/vrischmann/envconfig"
+)
+
+type BrokerClient interface {
+	Deprovision(instance internal.Instance) (string, error)
+}
+
+type Config struct {
+	Database storage.Config
+	Broker   broker.ClientConfig
+	DryRun   bool `envconfig:"default=true"`
+}
+
+type DeprovisionRetriggerService struct {
+	cfg             Config
+	filter          dbmodel.InstanceFilter
+	instanceStorage storage.Instances
+	brokerClient    BrokerClient
+}
+
+func main() {
+	log.SetFormatter(&log.JSONFormatter{})
+	log.Info("Starting deprovision retrigger job!")
+
+	// create and fill config
+	var cfg Config
+	err := envconfig.InitWithPrefix(&cfg, "APP")
+	fatalOnError(err)
+
+	if cfg.DryRun {
+		log.Info("Dry run only - no changes")
+	}
+
+	ctx := context.Background()
+	brokerClient := broker.NewClient(ctx, cfg.Broker)
+
+	// create storage connection
+	cipher := storage.NewEncrypter(cfg.Database.SecretKey)
+	db, conn, err := storage.NewFromConfig(cfg.Database, events.Config{}, cipher, log.WithField("service", "storage"))
+	fatalOnError(err)
+	svc := newDeprovisionRetriggerService(cfg, brokerClient, db.Instances())
+
+	err = svc.PerformCleanup()
+
+	fatalOnError(err)
+
+	log.Info("Deprovision retrigger job finished successfully!")
+
+	err = conn.Close()
+	if err != nil {
+		fatalOnError(err)
+	}
+
+	// do not use defer, close must be done before halting
+	err = cleaner.Halt()
+	fatalOnError(err)
+
+	time.Sleep(5 * time.Second)
+}
+
+func newDeprovisionRetriggerService(cfg Config, brokerClient BrokerClient, instances storage.Instances) *DeprovisionRetriggerService {
+	return &DeprovisionRetriggerService{
+		cfg:             cfg,
+		instanceStorage: instances,
+		brokerClient:    brokerClient,
+	}
+}
+
+func (s *DeprovisionRetriggerService) PerformCleanup() error {
+	notCompletelyDeletedFilter := dbmodel.InstanceFilter{DeletionAttempted: &[]bool{true}[0]}
+	instancesToDeprovisionAgain, _, _, err := s.instanceStorage.List(notCompletelyDeletedFilter)
+
+	if err != nil {
+		log.Error(fmt.Sprintf("while getting not completely deprovisioned instances: %s", err))
+		return err
+	}
+
+	if s.cfg.DryRun {
+		s.logInstances(instancesToDeprovisionAgain)
+		log.Infof("Instances to retrigger deprovisioning: %d", len(instancesToDeprovisionAgain))
+	} else {
+		deprovisioningAccepted, failuresCount := s.retriggerDeprovisioningForInstances(instancesToDeprovisionAgain)
+		log.Infof("Instances to retrigger deprovisioning: %d, accepted requests: %d, failed requests: %d", len(instancesToDeprovisionAgain), deprovisioningAccepted, failuresCount)
+	}
+
+	return nil
```

Comment:
Since the job itself works fine and iterates over the list of instances to deprovision I would prefer to leave exiting with 0 unless we crashed/panicked.